designs/CentrallyManagingNuGetPackageVersions-VSManagePackages.md

```diff
@@ -0,0 +1,414 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="PackageB" />
+  </ItemGroup>
+
+</Project>
+```
+
+### Update the version
+
+In this scenario, there is a `Directory.Packages.props` file which defines a version for `PackageA` and `PackageB`. There are two projects, `Project1` has a `PackageReference` to `PackageA`, and `Project2` has a `PackageReference` to `PackageB`. The customer updates `PackageA` to a different version. The `Directory.Packages.props` must be modified to contain the new version. All projects (two in this scenario) must be restored. `Project1` must be restored as it directly references `PackageA`. `Project2` must be restored, because `PackageB` might have a dependency on `PackageA`, and therefore the `PackageA`'s version needs to be pinned.
+
+#### Before
+
+`$(repo_root)\Directory.Packages.props`
+
+```xml
+<Project>
+  <ItemGroup>
+    <PackageVersion Include="PackageA" Version="1.2.3" />
+    <PackageVersion Include="PackageB" Version="4.5.6" />
+  </ItemGroup>
+</Project>
+```
+
+`$(repo_root)\Project1\Project1.csproj`
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="PackageA" />
+  </ItemGroup>
+
+</Project>
+```
+
+`$(repo_root)\Project2\Project2.csproj`
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="PackageB" />
+  </ItemGroup>
+
+</Project>
+```
+
+#### Gesture
+
+The customer updates `PackageA` to version `2.0.0` in Package Manager UI.
+
+#### After
+
+`$(repo_root)\Directory.Packages.props`
+
+```xml
+<Project>
+  <ItemGroup>
+    <PackageVersion Include="PackageA" Version="2.0.0" />
+    <PackageVersion Include="PackageB" Version="4.5.6" />
+  </ItemGroup>
+</Project>
+```
+
+`$(repo_root)\Project1\Project1.csproj`
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="PackageA" />
+  </ItemGroup>
+
+</Project>
+```
+
+`$(repo_root)\Project2\Project2.csproj`
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="PackageB" />
+  </ItemGroup>
+
+</Project>
+```
+
+## Requirements
+
+The requirements are defined in the [feature spec](https://github.com/NuGet/Home/wiki/Centrally-managing-NuGet-package-versions).
+
+Note that:
+
+* It's valid for some, not all, projects in the solution to use centrally managed package versions.
+* It's valid for different projects using central package versions to use different `Directory.Packages.props` files.
```

Comment:
I'd like to understand how the preview restore works in this situation. Think it's the one hole I see with this design.

Reply:
I remember David suggested not doing the preview...comes with its own issues ofc :)